crates/aptos/src/governance/mod.rs

```diff
@@ -183,3 +203,166 @@ impl std::fmt::Display for ProposalMetadata {
         )
     }
 }
+
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct ScriptHash {
+    hash: String,
+    bytecode: String,
+}
+
+#[derive(Parser)]
+pub struct PrepareScript {
+    /// Path to the Move package that contains the execution script
+    #[clap(long, parse(from_os_str))]
+    pub path: PathBuf,
+}
+
+#[async_trait]
+impl CliCommand<ScriptHash> for PrepareScript {
+    fn command_name(&self) -> &'static str {
+        "PrepareScript"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<ScriptHash> {
+        let build_config = BuildConfig {
+            additional_named_addresses: BTreeMap::new(),
+            generate_abis: false,
+            generate_docs: false,
+            ..Default::default()
+        };
+
+        let compiled_package = compile_move(build_config, self.path.as_path())?;
```

Comment:
Are we expecting a path to a dir or file? Does this command give a good error message if you pass in the wrong one? If not, let's check this before calling `compile_move`.

Reply:
Also let's add `.context` if possible.

Reply:
We relied on Move compilation tool. It reports errors. In general, proposers must create Move a package folder and put Move.toml. This is not different from `aptos move compile`.